engine/render/src/render/render.cpp

```diff
@@ -542,7 +580,52 @@ namespace dmRender
         }
     }
 
-    Result DrawRenderList(HRenderContext context, HPredicate predicate, HNamedConstantBuffer constant_buffer)
+    ///////////////////////////////////////////////////////////////////////////////////////////////////
+
+    static bool RenderListEntryEqFn(RenderListEntry* a, RenderListEntry* b)
+    {
+        return a->m_Dispatch == b->m_Dispatch;
+    }
+
+    static void SetVisibility(uint32_t count, RenderListEntry* entries, Visibility visibility)
+    {
+        for (uint32_t i = 0; i < count; ++i)
+        {
+            entries[i].m_Visibility = visibility;
+        }
+    }
+
+    static void FrustumCulling(HRenderContext context, const dmIntersection::Frustum& frustum)
+    {
+        DM_PROFILE(Render, "FrustumCulling");
+
+        uint32_t num_entries = context->m_RenderList.Size();
+        if (num_entries == 0)
+            return;
+
+        BatchIterator<RenderListEntry*> iter(num_entries, context->m_RenderList.Begin(), RenderListEntryEqFn);
+        while(iter.Next())
+        {
+            RenderListEntry* batch_start = iter.Begin();
+
+            const RenderListDispatch* d = &context->m_RenderListDispatch[batch_start->m_Dispatch];
+            if (!d->m_VisibilityFn)
+            {
+                SetVisibility(iter.Length(), iter.Begin(), dmRender::VISIBILITY_FULL);
```

Comment:
For render list entries with no culling function, default to `dmRender::VISIBILITY_FULL`